lib/api/autoreply.js

```diff
@@ -85,24 +115,90 @@ module.exports = (db, server) => {
                 }
             }
 
-            const r = await db.database.collection('autoreplies').updateOne({ user }, { $set: result.value }, { upsert: true });
+            result.value.created = new Date();
+            result.value.created.setMilliseconds(0);
+            result.value.created = result.value.created.toISOString();
+
+            const autoreplyData = await db.database
+                .collection('autoreplies')
+                .findOneAndUpdate({ user }, { $set: result.value }, { returnDocument: 'after', upsert: true });
+
+            server.loggelf({
+                short_message: `[AUTOREPLY] ${autoreplyData.lastErrorObject.upserted ? 'create' : 'update'}`,
```

Comment:
When chaining properties of an unknown data structure (maybe a future API update renames `lastErrorObject` to something else), use optional chaining: `autoreplyData.lastErrorObject?.upserted` (on this line and others as well)